src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurCashAdvancePdpFeedFileServiceImpl.java

```diff
@@ -51,32 +53,73 @@ public boolean createPdpFeedFileForValidatedDetailFileLines(ConcurRequestExtract
         return pdpFileSuccessfullyCreated;
     }
 
-    private PdpFeedFileBaseEntry buildPdpFeedBaseEntry(ConcurRequestExtractFile requestExtractFile) {
+    private PdpFeedFileBaseEntry buildPdpFeedBaseEntry(ConcurRequestExtractFile requestExtractFile, ConcurRequestExtractBatchReportData reportData) {
         int totalPdpDetailRecordsCount = 0;
         KualiDecimal totalPdpDetailRecordsAmount = KualiDecimal.ZERO;
+        int totalClonedCashAdvanceLinesCount = 0;
+        KualiDecimal totalClonedCashAdvanceLinesAmount = KualiDecimal.ZERO;
+        int totalCashAdvancesUseInExpenseReportCount = 0;
+        KualiDecimal totalCashAdvancesUseInExpenseReportAmount= KualiDecimal.ZERO;
+        int totalDuplicateCashAdvanceCount = 0;
+        KualiDecimal totalDuplicateCashAdvanceAmount = KualiDecimal.ZERO;
+        int totalNonCashAdvanceRecordsCount = 0;
+        KualiDecimal totalNonCashAdvanceRecordsAmount = KualiDecimal.ZERO;
+        int totalCashAdvancesNotProcessedDueToValidationErrorsCount = 0;
+        KualiDecimal totalCashAdvancesNotProcessedDueToValidationErrorsAmount = KualiDecimal.ZERO;
+        int totalProcessingAttemptedLineCount = 0;
+        KualiDecimal totalProcessingAttemptedLineAmount = KualiDecimal.ZERO;
```

Comment:
I'll let Daniela give her opinion on the matter.  Would it help have helper methods in the report data object that add an amount to a total, and increment the related counter?